engine/src/test/java/org/hibernate/validator/test/internal/constraintvalidators/hv/URLValidatorTest.java

```diff
@@ -290,10 +290,91 @@ private void runUrlContainerValidation(Validator validator, URLContainer contain
 	}
 
 	private void assertValidUrls(ConstraintValidator<URL, CharSequence> validator) {
+		//valid urls
 		assertTrue( validator.isValid( null, null ) );
+		assertTrue( validator.isValid( "ftp://abc.de", null ) );
+		assertTrue( validator.isValid( "http://foo.com/blah_blah", null ) );
+		assertTrue( validator.isValid( "http://foo.com/blah_blah/", null ) );
+		assertTrue( validator.isValid( "http://foo.com/blah_blah_(wikipedia)", null ) );
+		assertTrue( validator.isValid( "http://foo.com/blah_blah_(wikipedia)_(again)", null ) );
+		assertTrue( validator.isValid( "http://www.example.com/wpstyle/?p=364", null ) );
+		assertTrue( validator.isValid( "https://www.example.com/foo/?bar=baz&inga=42&quux", null ) );
+		assertTrue( validator.isValid( "http://✪df.ws/123", null ) );
+		assertTrue( validator.isValid( "http://userid:password@example.com:8080", null ) );
+		assertTrue( validator.isValid( "http://userid:password@example.com:8080/", null ) );
+		assertTrue( validator.isValid( "http://userid@example.com", null ) );
+		assertTrue( validator.isValid( "http://userid@example.com/", null ) );
+		assertTrue( validator.isValid( "http://userid@example.com:8080", null ) );
+		assertTrue( validator.isValid( "http://userid@example.com:8080/", null ) );
+		assertTrue( validator.isValid( "http://userid:password@example.com", null ) );
+		assertTrue( validator.isValid( "http://userid:password@example.com/", null ) );
+		assertTrue( validator.isValid( "http://142.42.1.1/", null ) );
+		assertTrue( validator.isValid( "http://142.42.1.1:8080/", null ) );
+		assertTrue( validator.isValid( "http://➡.ws/䨹", null ) );
+		assertTrue( validator.isValid( "http://⌘.ws", null ) );
+		assertTrue( validator.isValid( "http://⌘.ws/", null ) );
+		assertTrue( validator.isValid( "http://foo.com/blah_(wikipedia)#cite-1", null ) );
+		assertTrue( validator.isValid( "http://foo.com/blah_(wikipedia)_blah#cite-1", null ) );
+		assertTrue( validator.isValid( "http://foo.com/unicode_(✪)_in_parens", null ) );
+		assertTrue( validator.isValid( "http://foo.com/(something)?after=parens", null ) );
+		assertTrue( validator.isValid( "http://☺.damowmow.com/", null ) );
+		assertTrue( validator.isValid( "http://code.google.com/events/#&product=browser", null ) );
+		assertTrue( validator.isValid( "http://j.mp", null ) );
+		assertTrue( validator.isValid( "ftp://foo.bar/baz", null ) );
+		assertTrue( validator.isValid( "http://foo.bar/?q=Test%20URL-encoded%20stuff", null ) );
+		assertTrue( validator.isValid( "http://مثال.إختبار", null ) );
```

Comment:
I'm surprised the IDN URLs pass with java.net.URI. I used to think that they were not supported (I had to use galimatias to parse URIs with IDN). Did it change recently?